Framework/PythonInterface/mantid/plots/mantidaxes.py

```diff
@@ -268,9 +268,10 @@ def artists_workspace_has_errors(self, artist):
             if any([workspace.readE(i)[spec_num] != 0 for i in range(0, workspace.getNumberHistograms())]):
                 return True
         else:
-            workspace_index = workspace.getIndexFromSpectrumNumber(spec_num)
-            if any(workspace.readE(workspace_index) != 0):
-                return True
+            if spec_num is not None:
```

Comment:
Since this if-statement is the only thing inside the else it can be changed to `elif`.